### PR TITLE
Bp 473 redesign settings page

### PR DESCRIPTION
**Pull request by frederike-ramin**

### CODE REVIEW

File: src/GridView/GridView.js [hunk 1]
```diff
@@ -1,90 +1,88 @@
...
+	}
+
+	render() {
+		const allFetches = PromiseState.all([this.props.entity]);
+		const entity = this.props.entity.value;
+		const connectionIncomplete = super.render(allFetches);
+		if(connectionIncomplete) {
+			return connectionIncomplete;
+		}
+		const childEntityTypes = this.getChildEntityTypes(entity.TypeId, window.hierarchy);
```
Comment (hpibptlab) [inline src/GridView/GridView.js]: CRITICAL "getChildEntityTypes" expects 1 argument, but 2 were provided. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS930)

File: src/App.js [hunk 2]
```diff
@@ -14,4 +22,6 @@ class App extends Component {
     }
 }
 
-export default App;
+export default connect.defaults({fetch: ConnectionComponent.switchFetch})(props => ({
```
Comment (hpibptlab) [inline src/App.js]: MAJOR Remove the unused function parameter "props". [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: src/SettingsView/EntityMappingListItem.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        return (
+            <ListItem>
+                <Row>
+                    <Col md={10}>
+                        {this.props.eventType.Name} - {this.getEntityTypeName(this.props.mapping.EntityTypeId)}
+                    </Col>
+                    <Col md={2}>
+                        {/*TODO add functionality*/}
```
Comment (hpibptlab) [inline src/SettingsView/EntityMappingListItem.js]: INFO Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

File: src/SettingsView/EventType.js [hunk 4]
```diff
@@ -1,103 +1,150 @@
...
+		this.setState({mappingExpanded: expanded});
+	}
+	
+	handleEventTypeDeletion() {
+		this.props.deleteEventType(this.props.eventType);
+	}
+
+	getEventTypeHeaderButtons() {
+		return [
+			// TODO add functionality
```
Comment (hpibptlab) [inline src/SettingsView/EventType.js:42]: INFO Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

**Comment by hpibptlab** [comment]
SonarQube analysis reported 4 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[CRITICAL]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'
[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by hpibptlab** [comment]
SonarQube analysis reported 2 issues
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

[INFO]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO'

File: src/SettingsView/EntityMappingListItem.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+		}
+
+		return (
+			<ListItem>
+				<Row>
+					<Col md={10}>
+						{this.props.eventType.Name} - {this.getEntityTypeName(this.props.mapping.EntityTypeId)}
+					</Col>
+					<Col md={2}>
+						{/*TODO add functionality*/}
```
Comment (hpibptlab) [inline src/SettingsView/EntityMappingListItem.js:58]: INFO Complete the task associated to this TODO comment. [rule](http://bpt-lab.org/sonarqube/coding_rules#rule_key=javascript%3AS1135)

**Review by julianweise — APPROVED** [review]

**Pull request merged**